### ISSUE TITLE
New buzzwords With Innovative Gaming Moves Googles AI Becomes Go Grandmaster in 3 Days

### ISSUE THREAD

**Issue by postmillenial** [postmillenial/reading-list#1237]
With Innovative Gaming Moves, Google&rsquo;s AI Becomes Go Grandmaster in 3 Days<br>
http://ift.tt/2zf7zMs<br><img width="640" height="353" src="http://ift.tt/2zrpgJE" class="webfeedsFeaturedVisual wp-post-image wp-stateless-item" style="display: block; margin: auto; margin-bottom: 20px;" data-image-size="large" data-stateless-media-bucket="cdn.thenewstack.io" data-stateless-media-name="media/2017/11/967b4b85-alphago-ratings-2.jpg"><p>It seems like an eon ago, but it was only last year in this accelerating era of ever-smarter machines when an AI designed to beat the world&rsquo;s best human players at the ancient and complexly nuanced game of Go actually accomplished the feat. AlphaGo, the Go-playing AI created by <a href="https://deepmind.com/" target="_blank" class="ext-link" rel="external nofollow">DeepMind</a>, the London-based AI lab that is a subsidiary of <a href="http://bit.ly/2x5VLZD" class="ext-link" rel="external nofollow">Google,</a> surprised many expert observers last year when it <a href="http://ift.tt/2p9dkri" target="_blank" class="local-link">defeated the legendary player Lee Sedol</a>, as well as decisively winning all three games against Go world champion&nbsp;Ke Jie earlier this year, before being retired from competition.</p>
<p>But despite its apparent prowess, AlphaGo relied on human expertise to train it. The AI learned from over a dataset of over 100,000 master-level games, in addition to improvements gleaned from playing games many times against itself. Yet, the team that developed it says that by using data from past games may have inadvertently &ldquo;[imposed]<span class="current-selection">&nbsp;a ceiling&rdquo; on the AlphaGo&rsquo;s perfo</span><span class="current-selection">rmance.</span></p>
<p>Now, DeepMind says that it has developed an even stronger Go-playing AI &mdash; AlphaGo Zero &mdash; which is capable of learning to play master-level Go and beyond, and it learns on is own, without any human intervention at all.</p>
<p></p>
<h2>No Human Data</h2>
<p>According to the company&rsquo;s findings, which were recently published in <a href="http://ift.tt/2zqwdup" target="_blank" class="ext-link" rel="external nofollow"><em>Nature</em></a>, AlphaGo Zero started out versed only in the basic rules of Go and was able to reach the grandmaster level &mdash; in only <em>three days</em>.</p>
<p>It did so by skipping the human-generated training data that previous versions of AlphaGo utilized, namely, the game data from the thousands of games played by human amateurs and professionals, which were fed into the system. Instead, AlphaGo Zero learns to master the game by playing against itself, using a new form of reinforcement learning to effectively become its own teacher. In doing so, AlphaGo Zero is able to even beat its machine predecessors by 100 to 0, prompting the DeepMind team to call it &ldquo;arguably the strongest Go player in history.&rdquo;</p>
<p><img class="aligncenter size-full wp-image-3412434" src="http://ift.tt/2zdLoWZ" width="1000" height="578"></p>
<h2>
<em>Tabula Rasa</em> Learning</h2>
<p>There are some interesting differences between AlphaGo and AlphaGo Zero which make the latest iteration so powerful and crushingly efficient. As the DeepMind researchers explain in their paper,&nbsp;<a href="http://ift.tt/2zqwdup" target="_blank" class="ext-link" rel="external nofollow"><em><span class="current-selection">Mast</span><span class="current-selection">e</span><span class="current-selection">ring th</span><span class="current-selection">e g</span><span class="current-selection">am</span><span class="current-selection">e of Go wi</span><span class="current-selection">th</span><span class="current-selection">o</span><span class="current-selection">u</span><span class="current-selection">t</span> <span class="current-selection">h</span><span class="current-selection">um</span><span class="current-selection">an kn</span><span class="current-selection">o</span><span class="current-selection">w</span><span class="current-selection">l</span><span class="current-selection">edge</span></em></a><span class="current-selection">, the initial versions of AlphaGo featured two deep neural networks &mdash; one &ldquo;policy network&rdquo; that selects the next move and a &ldquo;value network&rdquo; that evaluates and predicts the winner of the game from each possible position. The value network was tasked with predicting the winner of the games played by the policy network against itself.</span></p>
<p><span class="current-selection">Once these neural networks are trained, they are combined with a Monte Carlo tree search &mdash; which uses repeated random sampling algorithms to simulate the risks, possible outcomes and probabilities of various&nbsp;moves and counter-moves. The policy network is used to select the moves with the best outcomes, while the value network evaluates the positions relative to the rest of the tree search. This method is what helped AlphaGo decide which move to make, often surprising its human competitor with unexpected moves that hinted at some kind of ghost &ldquo;imagination&rdquo; in the machine.</span></p>
<p>AlphaGo Zero, however, only uses one neural network, combining both policy and value networks into one so that it could be trained more efficiently. This was used in conjunction with a new form of &ldquo;self-play&rdquo; reinforcement learning algorithm that incorporates a lookahead search function inside the training loop, meaning the system was able to learn more quickly and consistently. This new system also did away with the random &ldquo;rollout&rdquo; games as simulated by the aforementioned&nbsp;<span class="current-selection">Monte Carlo method in previous AlphaGo incarnations, instead relying on its own powerful neural network to analyze moves.</span></p>
<p>&ldquo;As [AlphaGo Zero] plays, the neural network is tuned and updated to predict moves, as well as the eventual winner of the games,&rdquo; explained the researchers. &ldquo;This updated neural network is then recombined with the search algorithm to create a new, stronger version of AlphaGo Zero, and the process begins again. In each iteration, the performance of the system improves by a small amount, and the quality of the self-play games increases, leading to more and more accurate neural networks and ever stronger versions of AlphaGo Zero.&rdquo;</p>
<p>The end result: by removing humans from the equation completely, the system is &ldquo;no longer constrained by the limits of human knowledge&rdquo; and can learn to start from a blank slate, or <em>tabula rasa</em>, each time from the most powerful player in the world: itself.</p>
<p>Even more surprising is the fact that once AlphaGo Zero surpassed the grandmaster level achieved by the previous strongest version after about 40 days of&nbsp;self-training and millions of games, it then seemed to develop completely unconventional and unanticipated maneuvers &mdash; or &ldquo;discovering new knowledge&rdquo; as the team describes it. AlphaGo Zero even went so far to discard certain traditional <a href="http://ift.tt/1W2pqII" target="_blank" class="ext-link" rel="external nofollow">opening moves and sequences</a> in Go, painstakingly developed by humans during the 2,500-year history of the game, in favor of its own innovations.</p>
<p><img class="aligncenter size-full wp-image-3412436" src="http://ift.tt/2zdLpKx" width="700" height="340"></p>
<p>There are many potential implications here: the foremost being that the algorithm behind such a self-learning system might very well be one piece of the puzzle in developing a <a href="http://ift.tt/1tjmfRL" target="_blank" class="ext-link" rel="external nofollow">strong, general AI</a> that&rsquo;s capable of mastering many different areas, just as humans might &mdash; as opposed to the so-called <a href="http://ift.tt/1r6dQRE" target="_blank" class="ext-link" rel="external nofollow">weak AI</a> that&rsquo;s only proficient at one, specific task. Yet another advantage would be eliminating the need for large data sets as training fodder for such systems, as these are often costly or difficult to obtain, or in some cases, non-existent.</p>
<p>&ldquo;<em>Tabula rasa</em> learning is extremely important to our goals and ambitions at DeepMind,&rdquo; said&nbsp;David Silver, AlphaGo Zero&rsquo;s lead programmer. &ldquo;The reason is that if you can achieve <em>tabula rasa</em> learning, you have an agent that can be transplanted from the game of Go to any other domain. You untie yourself from the specifics of the domain you are in and you come up with an algorithm that is so general that it can be applied anywhere.&rdquo;</p>
<p>If the innovative leaps and bounds that AlphaGo Zero was able to make in such a short time are any kind of indicator, such &ldquo;general purpose&rdquo; algorithms could potentially accelerate the discovery of new, unexpected knowledge in diagnosing and curing intractable diseases, solving the growing climate crisis or even help us gain some scientific insights into the mysteries of the universe. It&rsquo;s still a far stretch from here to that place in time, but such an artificially intelligent future seems even&nbsp;more possible now.</p>
<p class="attribution"><a href="http://bit.ly/2x5VLZD" class="ext-link" rel="external nofollow">Google</a> is a sponsor of The New Stack.</p>
<p class="attribution">Images: DeepMind</p>
<p>The post <a rel="nofollow" href="http://ift.tt/2zf7zMs">With Innovative Gaming Moves, Google&rsquo;s AI Becomes Go Grandmaster in 3 Days</a> appeared first on <a rel="nofollow" href="https://thenewstack.io">The New Stack</a>.</p>
<br><br>
via Blog &ndash; The New Stack https://thenewstack.io<br>
November 02, 2017 at 02:20PM
